Refactor postponed processing in process_tv, cleaned up a bit

medariox opened the pull request:
- [x] PR is based on the DEVELOP branch
- [x] Don't send big changes all at once. Split up big PRs into multiple smaller PRs that are easier to manage and review
- [x] Read the [contribution guide](https://github.com/pymedusa/Medusa/blob/master/.github/CONTRIBUTING.md)

fernandog commented on medusa/process_tv.py:
```diff
@@ -627,7 +616,7 @@ def subtitles_enabled(*args):
                 continue
 
             try:
-                parse_result = NameParser().parse(name, cache_result=True)
```
@medariox why remove this?

medariox replied:
Yes

fernandog commented:
@medariox is better to tell me what you changed or just moved core around ?

medariox commented:
@fernandog 
I moved the postpone processing logic out of process_media, because it was getting too big. Removed some useless code in prepare_files and cleaned up a bit. I just want to make sure postpone works as expected. I tested it ofc, but since you use it constantly, it would be nice if you could verify as well.

fernandog commented:
Did a quick test and seems to work.
I will add to production and let you know

fernandog commented on medusa/process_tv.py:
```diff
@@ -573,7 +573,6 @@ def _process_postponed(self, processor, path, video, ignore_subs):
             self._log('Subtitles check was disabled for this episode in manual post-processing. '
                       'Continuing the post-processing of this file: {0}'.format(video))
 
-        self.postpone_processing = False
```
@medariox in my tests this was needed. I didn't understood why but it was needed

fernandog replied:
@medariox in my tests this was needed. I didn't understood why but it was needed

fernandog replied:
@medariox 
I think because when its set to True it never goes back to False.

medariox commented:
```
Destination folder for this episode: /home/marx/Documents/TV Shows/Show/S02
Found the following associated files for /home/marx/Downloads/complete/Show.S02E06.HDTV.XviD-FUM[ettv]/Show.S02E06.HDTV.XviD-FUM[ettv].mkv: set([u'/home/marx/Downloads/complete/Show.S02E06.HDTV.XviD-FUM[ettv]/Show.S02E06.HDTV.XviD-FUM[ettv].en.srt'])
Hard linking file from /home/marx/Downloads/complete/Show.S02E06.HDTV.XviD-FUM[ettv]/Show.S02E06.HDTV.XviD-FUM[ettv].mkv to /home/marx/Documents/TV Shows/Show/S02/Show.S02E06.HDTV.XviD-FUM.mkv
Moving file from /home/marx/Downloads/complete/Show.S02E06.HDTV.XviD-FUM[ettv]/Show.S02E06.HDTV.XviD-FUM[ettv].en.srt to /home/marx/Documents/TV Shows/Show/S02/Subs/Show.S02E06.HDTV.XviD-FUM.en.srt
```
It's obvious just by looking at the code, as what I have changed has no influence on that.

medariox commented:
That comment didn't make sense, especially not in that code part. If you want me to fix it (I can do it, I don't have to as I didn't break it), you will have to explain to explain in proper english and with a different tone, exactly what it is supposed to do and why.
